src/frontend/include/yaksi.h

```diff
@@ -42,6 +42,30 @@ typedef enum {
 struct yaksi_type_s;
 struct yaksi_request_s;
 
+/*
+ * The type handle is divided into the following parts:
+ *
+ *    32 bits -- unused
+ *    32 bits -- type object id
+ */
+#define YAKSI_TYPE_UNUSED_BITS                  (32)
+#define YAKSI_TYPE_OBJECT_ID_BITS               (32)
+
+#define YAKSI_TYPE_GET_OBJECT_ID(handle) \
+    (handle << (64 - YAKSI_TYPE_OBJECT_ID_BITS) >> (64 - YAKSI_TYPE_OBJECT_ID_BITS))
```

Comment:
I recommend `(handle) << ...` rather than `handle <<` to avoid an operator precedence bug.

Reply:
Fixed.